cmake/modules/hunter_pkgconfig_export_target.cmake

```diff
@@ -51,37 +46,53 @@ function(hunter_pkgconfig_export_target PKG_CONFIG_MODULE)
   set(link_libs)
 
   hunter_status_debug(
-      "PKG_CONFIG_MODULE ${PKG_CONFIG_MODULE} LDFLAGS: ${${PKG_CONFIG_MODULE}_LDFLAGS}"
-  )
-  string(COMPARE NOTEQUAL
-      "${${PKG_CONFIG_MODULE}_LDFLAGS}"
-      ""
-      has_ldflags
+      "PKG_CONFIG_MODULE ${PKG_CONFIG_MODULE} LINK_LIBRARIES: ${${PKG_CONFIG_MODULE}_LINK_LIBRARIES}"
   )
-  if(has_ldflags)
-    list(APPEND link_libs ${${PKG_CONFIG_MODULE}_LDFLAGS})
+  if(NOT "${${PKG_CONFIG_MODULE}_LINK_LIBRARIES}" STREQUAL "")
+    # Get and set the libraries and their absolute paths
+    list(APPEND link_libs ${${PKG_CONFIG_MODULE}_LINK_LIBRARIES})
   endif()
 
   hunter_status_debug(
-      "PKG_CONFIG_MODULE ${PKG_CONFIG_MODULE} LDFLAGS_OTHER: ${${PKG_CONFIG_MODULE}_LDFLAGS_OTHER}"
+      "PKG_CONFIG_MODULE ${PKG_CONFIG_MODULE} LIBRARY_DIRS: ${${PKG_CONFIG_MODULE}_LIBRARY_DIRS}"
   )
-  string(COMPARE NOTEQUAL
-      "${${PKG_CONFIG_MODULE}_LDFLAGS_OTHER}"
-      ""
-      has_ldflags_other
+  if(NOT "${${PKG_CONFIG_MODULE}_LIBRARY_DIRS}" STREQUAL "")
+    # Get only the paths of the libraries without '-L' and set with the '-L'
+    foreach(_dir ${${PKG_CONFIG_MODULE}_LIBRARY_DIRS})
+      set(link_dir -L${_dir})
+      list(APPEND link_libs ${link_dir})
+    endforeach()
+  endif()
+
+  hunter_status_debug(
+      "PKG_CONFIG_MODULE ${PKG_CONFIG_MODULE} LIBRARIES: ${${PKG_CONFIG_MODULE}_LIBRARIES}"
   )
-  if(has_ldflags_other)
-    list(APPEND link_libs ${${PKG_CONFIG_MODULE}_LDFLAGS_OTHER})
+  if(NOT "${${PKG_CONFIG_MODULE}_LIBRARIES}" STREQUAL "")
+    # Get and set only the libraries (without the '-l')
+    list(APPEND link_libs ${${PKG_CONFIG_MODULE}_LIBRARIES})
   endif()
 
-  # No need to treat the pkg-config module's _LIBRARY_DIRS and _LIBRARIES
-  # as they are already included in LD_FLAGS
-  string(COMPARE NOTEQUAL
-      "${link_libs}"
-      ""
-      has_link_libs
+  hunter_status_debug(
+      "PKG_CONFIG_MODULE ${PKG_CONFIG_MODULE} LDFLAGS_OTHER: ${${PKG_CONFIG_MODULE}_LDFLAGS_OTHER}"
   )
-  if(has_link_libs)
+  if(NOT "${${PKG_CONFIG_MODULE}_LDFLAGS_OTHER}" STREQUAL "")
+    # Get all other linker flags
```

Comment:
will this code do the same:
```cmake
# turn "-framework;A;-framework;B" into "-framework A;-framework B"
string(REPLACE "-framework;" "-framework " ldflags_other "${${PKG_CONFIG_MODULE}_LDFLAGS_OTHER}")
```

?

Reply:
Hmm, maybe... let me think, we take the list and works like a string and replaces `;` with ` ` ... hm should also work.

Reply:
Done